prometheus-client/src/Prometheus/Metric/Summary.hs

```diff
@@ -9,187 +8,94 @@ module Prometheus.Metric.Summary (
 ,   observe
 ,   observeDuration
 ,   getSummary
-
-,   dumpEstimator
-,   emptyEstimator
-,   Estimator (..)
-,   Item (..)
-,   insert
-,   compress
-,   query
-,   invariant
 ) where
 
 import Prometheus.Info
 import Prometheus.Metric
 import Prometheus.Metric.Observer
 import Prometheus.MonadMonitor
 
-import qualified Control.Concurrent.STM as STM
+import Control.Concurrent.MVar
 import Control.DeepSeq
+import Control.Monad
 import Control.Monad.IO.Class
+import Control.Monad.Primitive
 import qualified Data.ByteString.UTF8 as BS
-import Data.Foldable (foldr')
-import Data.Int (Int64)
-import Data.Monoid ((<>))
 import qualified Data.Text as T
+import DataSketches.Quantiles.RelativeErrorQuantile
+import qualified DataSketches.Quantiles.RelativeErrorQuantile as ReqSketch
+import Data.Maybe (mapMaybe)
+import Prelude hiding (maximum)
+import qualified Prelude
+import Data.Word
+
+data Summary = MkSummary
+  { reqSketch :: MVar (ReqSketch (PrimState IO))
+  , quantiles :: [Quantile]
+  }
 
+instance NFData Summary where
+  rnf (MkSummary a b) = a `seq` b `seq` ()
```

Comment:
`seq` for `b` doesn't look right here - don't you need to deepseq the entire list?

Reply:
d'oh, should be fixed now.